### DISCUSSION TITLE
CookieConsent & Wordpress wp_enqueue_script

### DISCUSSION THREAD

**Discussion by lombrone** [orestbida/cookieconsent#754] [Q&A]
Hello,

I installed the script in wordpress with something like this in function.php :

```
add_action( 'wp_footer', 'load_scripts' );
function load_scripts() {
   wp_enqueue_script_module('cookie-consent', get_template_directory_uri() . '/05-assets/js/cookieconsent-config.js');
   wp_enqueue_script('script', get_template_directory_uri() . '/js/my-common.js'); 
}
```

It's working, but i needed to add `type` and `data-category` to my custom script, so i tried to use `script_loader_tag`.
But once i accept cookie, my script don't work even if `text/plain` has been removed automatically after accept.

```
add_filter( 'script_loader_tag', 'wpse397773_change_script_tags', 10, 3 );
function wpse397773_change_script_tags( $tag, $handle, $src ) {
    if ( 'script' == $handle ) {
        $tag = '<script type="text/plain" data-category="necessary" src="' . $src . '" id="' . $handle . '-js"></script>';
    }
    return $tag;
}
```

It's seem `script_loader_tag` don't manage `text/plain` perfectly. If someone who knows Wordpress have a tip.

**Comment by rocky2b** [comment]
i'm working in a WP environment with cookie consent in a custom plugin.

https://developer.wordpress.org/reference/hooks/script_loader_tag/

It just overrides the HTML markup that gets generated on the frontend. 

> But once i accept cookie, my script don't work

My guess is that the script order/priority is off. 

This is something I can explore and gave a better answer to when I get to it in my project, in the next 2-3 weeks.
